### ISSUE TITLE
possible error formatting error message in security module [JIRA: RIAK-1802]

### ISSUE THREAD

**Issue by marianoguerra** [basho/riak_core#706]
hi, I think this is an error, if I get a confirmation I would be more than happy to provide a pull request.

the issue starts when a call like this:

```
7> riak_core_security:check_permissions({"get", any}, Ctx).     
** exception error: bad argument
     in function  unicode:characters_to_binary/3
    called as unicode:characters_to_binary(any,utf8,utf8)
     in call from riak_core_security:check_permission/2 (src/riak_core_security.erl, line 380)

8> riak_core_security:check_permissions({"get", <<"foo">>}, Ctx).
{false,<<"Permission denied: User 'admin' does not have 'get' on foo">>,
       {context,<<"admin">>,
            [...],
            {1424,292313,620773}}}
```

bucket2iolist seems to be only called from the line 380 here:

https://github.com/basho/riak_core/blob/develop/src/riak_core_security.erl#L380

and it fails because instead of translating the atom any to something like "*" or "any" it calls unicode:characters_to_binary(Bucket, utf8, utf8) which fails.

should the fix be matching any and returning a string representation of it?

**Comment by marianoguerra** [comment]
proposed fix:

note, it doesn't replace Bucket being any in case the parameter is {Type, Bucket}, I haven't seen that case, but adding a case for that is easy

```
@@ -1325,6 +1325,8 @@ bucket2bin({Type, Bucket}) ->
 bucket2bin(Name) ->
     unicode:characters_to_binary(Name, utf8, utf8).

+bucket2iolist(any) ->
+    <<"*">>;
 bucket2iolist({Type, Bucket}) ->
     [unicode:characters_to_binary(Type, utf8, utf8), "/",
      unicode:characters_to_binary(Bucket, utf8, utf8)];
```

**Comment by marianoguerra** [comment]
here is a failing test case and a proposed fix:

https://github.com/marianoguerra/riak_core/tree/fix-issue-706

comparison here:

https://github.com/marianoguerra/riak_core/compare/fix-issue-706

I can do a pull request if it fits your workflow

**Comment by jcapricebasho** [comment]
comment for jira

**Comment by lukebakken** [comment]
@marianoguerra - Did you run into this during normal Riak usage or only from a `riak attach` session?

**Comment by Basho-JIRA** [comment]
Please see this pull request:

https://github.com/basho/riak_kv/pull/1116 

**_[posted via JIRA by Luke Bakken]**_

**Comment by Basho-JIRA** [comment]
[~jmeredith], can you get someone to review this and get added to the next Riak release?  The "Get Preflist" function will not work with Riak security enabled without this fix.  Thanks. 

**_[posted via JIRA by Derek Somogyi]**_

**Comment by Basho-JIRA** [comment]
This issue is currently holding up the PHP 2.1 release.  It appears as though the change/fix will be included in the 2.1.2 release.  I added that tag to the Labels field. 

**_[posted via JIRA by Derek Somogyi]**_

**Comment by Basho-JIRA** [comment]
Merged into 2.1 on July 7. 

**_[posted via JIRA by Douglas Rohrer]**_
